### ISSUE TITLE
Wrap DynSLAM as a ROS node 

### ISSUE THREAD

**Issue by rnunziata** [AndreiBarsan/DynSLAM#38]
Have you considered make this a ROS node so that depth, and segment classification can be integrated via ROS messages. This would allow switching between different implementation of these components, and would be language agnostic.

**Comment by AndreiBarsan** [comment]
Yes, I have. You are correct; those would be very neat advantages, but it would take a bit of time to refactor everything. It's on my todo list, though. And once again, very sorry about the slow replies, it's a busy period here and I'm also CUDA-less. :(

**Comment by AndreiBarsan** [comment]
Note to self: Peidong suggested ROS2 should be used (http://wiki.ros.org/ROS/Tutorials/WritingPublisherSubscriber%28c%2B%2B%29).

**Comment by 0dmf0** [comment]
> Yes, I have. You are correct; those would be very neat advantages, but it would take a bit of time to refactor everything. It's on my todo list, though. And once again, very sorry about the slow replies, it's a busy period here and I'm also CUDA-less. :(

Have you already make this a ROS node?

**Comment by AndreiBarsan** [comment]
Sorry, I haven't added ROS support and I don't think I'll have time in the foreseeable future. I'd be more than happy to review PRs adding that if someone does it!

**Comment by 0dmf0** [comment]
Do you know how to make a KITTI data with a rosbag which has rgb and depth message?



发自我的iPhone


------------------ Original ------------------

**Comment by AndreiBarsan** [comment]
ETH Zurich seem to have a utility for building rosbags from KITTI sequences here: https://github.com/ethz-asl/kitti_to_rosbag

However, since KITTI doesn't come with depth (apart from the LiDAR), you'd need to compute that separately, and modify the code above to include one additional depth or disparity image in each frame.

If you want to, you can use the DispNet also used by this project, or if you want to upgrade to newer tech (DispNet is almost 5 years old), I'd recommend one of my colleague's awesome DeepPruner: https://github.com/uber-research/DeepPruner (MIT licensed)

**Comment by 0dmf0** [comment]
> ETH Zurich seem to have a utility for building rosbags from KITTI sequences here: https://github.com/ethz-asl/kitti_to_rosbag
> 
> However, since KITTI doesn't come with depth (apart from the LiDAR), you'd need to compute that separately, and modify the code above to include one additional depth or disparity image in each frame.
> 
> If you want to, you can use the DispNet also used by this project, or if you want to upgrade to newer tech (DispNet is almost 5 years old), I'd recommend one of my colleague's awesome DeepPruner: https://github.com/uber-research/DeepPruner (MIT licensed)

Well, so can you explain clearly how to use another KITTI dataset? I don't know how to process another dataset.

**Comment by 0dmf0** [comment]
Or can you provide another KITTI dataset which has already been processed?
Thanks

**Comment by AndreiBarsan** [comment]
Do you mean pre-processed for DynSLAM or pre-processed for ROS? 
For DynSLAM there are tools in this repo (MNC and dispnet-flownet-docker in `preprocessing/`) do do it for KITTI data. For ROS I am not sure; I am not familiar with ROSbags.

**Comment by 0dmf0** [comment]
> Do you mean pre-processed for DynSLAM or pre-processed for ROS?
> For DynSLAM there are tools in this repo (MNC and dispnet-flownet-docker in `preprocessing/`) do do it for KITTI data. For ROS I am not sure; I am not familiar with ROSbags.

Everytime I pre-process KITTI data, it shows:
dmf@dmf-GS63-7RE:/media/dmf/丁/DynSLAM$ scripts/preprocess-sequence.sh kitti-tracking data/kitti/tracking training 6Failure: Invalid directory structure for the tracking dataset. 
Expected directory structure:
		├── training
		│   ├── calib
		│   ├── image_02
		│   │   ├── 0000
		│   │   └── etc.
		│   ├── image_03
		│   │   ├── 0000
		│   │   └── etc.
		│   └── velodyne
		│       ├── 0000
		│       └── etc.
		└── testing
			└── <same as training>.
Do you know how to solve this?

**Comment by AndreiBarsan** [comment]
What does your dir structure look like?

**Comment by 0dmf0** [comment]
![图片](https://user-images.githubusercontent.com/60035642/80867912-7d7ed280-8cc9-11ea-8d48-ed0e79373367.png)
is it wrong?

**Comment by 0dmf0** [comment]
/media/dmf/丁/2020-05-02 23-05-35屏幕截图.png

**Comment by AndreiBarsan** [comment]
Seems like the folder is called `training 1` instead of `training`. Maybe
that's the issue?

**Comment by 0dmf0** [comment]
bash:./preprocess-sequence.sh: no such file
Do you know why?
